### DISCUSSION TITLE
`client:*` Directives for Frameworks

### DISCUSSION THREAD

**Discussion by joezimjs** [withastro/roadmap#199] [Proposal]
For me, I'd prefer to not need to know how to create components in two different frameworks' syntaxes: Astro and Vue. I've been using Vue for a long time and would prefer to keep as much of my app using Vue as possible. The one thing that prevents this is my desire to keep most of the code back-end-only, but the moment I use a Vue component, I can no longer use the `client:` directives to mark what can run on client, so I either have to make the whole Vue component tree run on the client, none of it run on the client, or I need to use Astro components for anything server-side.

So, are there any plans to make `client` directives that work within Vue components to mark child components as client-side? If not, is there a reason to hold back?

**Comment by iankduffy** [comment]
I was also wondering if this was possible or if in react we could do the 'use client' like in nextJS https://beta.nextjs.org/docs/rendering/server-and-client-components#client-components

**Comment by florian-lefebvre** [comment]
There are no plans to have client directives inside framework components because we do not control the rendering inside UI frameworks, and I don't think `client:xxx` is a valid prop for some of them. This is a known limitation
